lib/puppet/reports/store.rb

```diff
@@ -32,7 +32,11 @@ def process
     file = File.join(dir, name)
 
     begin
-      Puppet::Util.replace_file(file, 0640) do |fh|
+      # Mode 640 has known issues on Windows when the owner and group of the report file are SYSTEM
+      # which can occur when Puppet is running on a service.  Instead by passing nil we can use the
+      # the file system inherited permissions, which are set by the Puppet Agent package.
```

Comment:
minor - `the the`

Can Linux use default perms too - or will umask get in our way?

Reply:
Unsure, so chose not to change it

Reply:
Fixed typo

Reply:
Note that passing in `nil` will make replace_file use `0644` https://github.com/puppetlabs/puppet/blob/2d00ee60ea14f774c027a815e3703861988fc74d/lib/puppet/util.rb#L560